logger/logger.go

```diff
@@ -91,11 +92,33 @@ SetLoggingLevel dynamically sets the logging level at runtime. See https://githu
 for possible logging levels.
 */
 func SetLoggingLevel(level string) {
+	mutex.Lock()
+	defer mutex.Unlock()
```

Comment:
@ameowlia 
I added the Mutex here since there are a few different ways the logging level can get changed.